lib/x11utils.pm

```diff
@@ -68,6 +68,17 @@ all possible options should be handled within loop to get unlocked desktop
 =cut
 sub ensure_unlocked_desktop {
     my $counter = 10;
+
+    # check if the screenbuffer was not updated and record a softfailure in that case
+    if (check_var('DESKTOP', 'gnome') && check_screen('screenlock')) {
```

Comment:
The negation is still missing:

```suggestion
    if (check_var('DESKTOP', 'gnome') && !check_screen('screenlock')) {
```

Reply:
As mentioned above I cannot follow that logic

Reply:
> This seems not logic to me, if we have a lockscreen and press shift we should not get another lockscreen

`check_screen('screenlock')` will return "true" if the screenlock if found.
If screenlock is found, the bug didn't happened. -> nothing to do

If screenlock is not found, it may be the bug. -> workaround

Reply:
And of course, if you have lockscreen and press shift (you can try yourself on your laptop/PC) you still get the lockscreen.